var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -368,35 +380,43 @@ def patch(self):
 
     @property
     def common_config_args(self):
+        spec = self.spec
+        version = self.version
+
         # incomplete list is here http://doc.qt.io/qt-5/configure-options.html
         config_args = [
             '-prefix', self.prefix,
             '-v',
             '-opensource',
-            '-{0}opengl'.format('' if '+opengl' in self.spec else 'no-'),
-            '-{0}'.format('debug' if '+debug' in self.spec else 'release'),
+            '-{0}opengl'.format('' if '+opengl' in spec else 'no-'),
+            '-{0}'.format('debug' if '+debug' in spec else 'release'),
             '-confirm-license',
             '-optimized-qmake',
             '-no-pch',
         ]
 
-        if self.spec.variants['freetype'].value == 'spack':
-            config_args.extend([
-                '-system-freetype'
-            ])
-            config_args.extend(
-                self.spec['freetype'].headers.include_flags.split()
-            )
+        def use_spack_dep(spack_pkg, qt_name=None):
+            pkg = spec[spack_pkg]
+            config_args.append('-system-' + (qt_name or spack_pkg))
+            if not pkg.external:
+                config_args.extend([
+                    pkg.libs.search_flags,
+                    pkg.headers.include_flags
+                ])
+
+        if '+gui' in spec:
+            use_spack_dep('freetype')
             if not MACOS_VERSION:
                 config_args.append('-fontconfig')
-
-        elif self.spec.variants['freetype'].value == 'qt':
-            config_args.append('-qt-freetype')
+            elif version < Version('5.12'):
+                # Linux-only QT5 dependencies
+                config_args.append('-system-xcb')
```

Comment:
Isn't this only not available anymore in 5.15?

Reply:
You're right, it was a typo. Good catch!